### ISSUE TITLE
Re-sync fails after foreign key/table removal

### ISSUE THREAD

**Issue by tjmcewan** [nocodb/nocodb#3033]
```
Node: **v12.22.12**
Arch: **x64**
Platform: **linux**
Docker: **false**
Database: **pg**
ProjectOnRootDB: **false**
RootDB: **pg**
PackageVersion: **0.92.4**
```

**Steps To Reproduce**

1. Import a db with multiple tables containing foreign keys
2. Delete one of the referenced foreign tables (& the corresponding foreign keys)
3. Reload the project metadata & notice the red rows (removed constraint(s), removed reference)
4. Try to sync
5. UI may show error, if so observe the error below (from Heroku logs, in my case):

```
O :: error: delete from "nc_columns_v2" where "id" = $1 - update or delete on table "nc_columns_v2" violates foreign key constraint "nc_col_relations_v2_fk_child_column_id_foreign" on table "nc_col_relations_v2"
```

I'm assuming it's due to the attempted deletion order of the column information, hence *may* show error.

(My "solution" was just to delete the project and re-add it; I've only just started using nocodb.)

**Comment by pgeliebter** [comment]
I'm getting the same error after attempting to resync after renaming tables in my Database

**Comment by o1lab** [comment]
@pranavxc

**Comment by pranavxc** [comment]
Could be related to the order of syncing the changes. I will have a look and thanks for sharing the error details.

**Comment by pranavxc** [comment]
@pgeliebter @tjmcewan - Fixed the above sync issue in following PR #3611. It will be helpful if you guys can test the following test build.

```sh
docker run -d -p 8888:8080 nocodb/nocodb-timely:0.96.4-pr-3611-20220913-1730
```
